src/components/hooks/useActivityStream.js

```diff
@@ -1,22 +1,45 @@
-import {useState, useEffect} from 'react';
-import {merge} from 'rxjs';
+import {useEffect, useReducer} from 'react';
+
+const APPEND_ACTIVITIES = 'append_activities';
+
+/**
+ * Returns a new array of activityIDs based on the given action.
+ * If no action is passed, it will return the same activityIDs
+ * array without any changes.
+ *
+ * @param {Array.<string>} activityIDs activityIDs associated to the room
+ * @param {object} action action to apply to given activityIDs
+ * @returns {Array.<string>} New activityIDs associated to the room
+ */
+function reducer(activityIDs, action) {
+  let newActivityIDs = [];
+
+  switch (action.type) {
+    case APPEND_ACTIVITIES:
+      newActivityIDs = [...activityIDs].concat(action.payload);
```

Comment:
Is the array deconstruction necessary here since `activityIDs` is already an array?

Reply:
no it's not.